### PR TITLE
Hash and prune copied CSVs in olm-operator.

### PR DESCRIPTION
**Pull request by benluddy**
OLM doesn't need to maintain a local cache of complete copied
ClusterServiceVersion objects because it also holds the source of
truth.

### CODE REVIEW

File: pkg/controller/operators/olm/operatorgroup.go [hunk 1]
```diff
@@ -641,13 +644,36 @@ func (a *Operator) ensureCSVsInNamespaces(csv *v1alpha1.ClusterServiceVersion, o
 	logger := a.logger.WithField("opgroup", operatorGroup.GetName()).WithField("csv", csv.GetName())
 
 	targetCSVs := make(map[string]*v1alpha1.ClusterServiceVersion)
+
+	copyPrototype := v1alpha1.ClusterServiceVersion{
```
Comment (ecordell) [inline pkg/controller/operators/olm/operatorgroup.go:648]: I don't have a strong preference here, but I think you could calculate the "prototype" in copyableCSVHash. that would make me more confident that we're hashing the same thing here as in the listwatcher

**Comment by benluddy** [comment]
/hold

Will need a rebase once https://github.com/operator-framework/operator-lifecycle-manager/pull/2282 merges -- that dependency bump fixes a bug in client-go/testing that affects this change.

**Review by dinhxuanvu — APPROVED** [review]
/lgtm

**Pull request merged**